homeassistant/helpers/debounce.py

```diff
@@ -117,8 +117,6 @@ async def _handle_timer_finish(self) -> None:
             except Exception:  # pylint: disable=broad-except
                 self.logger.exception("Unexpected exception from %s", self.function)
 
-            self._schedule_timer()
```

Comment:
Reading the linked PR, I think the idea is that we want to stop any call from executing within the cooldown. By removing this schedule timer call we will allow a manual call directly after a scheduled call, bypassing the cooldown.

Reply:
Yes I came to the same conclusion.
It looks like what I thought was an endless loop is actually just a single "extra run".
I'll see if I can improve...

Reply:
I have adjusted the code, to at least remove the unnecessary call to `async_create_task`

Another possibility is to keep hold of the "next authorized run time" and compare "now" with "last run time + cooldown" but I worry this would cause unnecessary computing.
It would also probably cause a lot of tests to fail.